presto-cli/src/main/java/com/facebook/presto/cli/Query.java

```diff
@@ -208,15 +214,28 @@ private void pageOutput(OutputFormat format, List<String> fieldNames)
 
         try (Writer writer = createWriter(Pager.create());
                 OutputHandler handler = createOutputHandler(format, writer, fieldNames)) {
-            handler.processRows(client);
+            try {
+                handler.processRows(client);
+            }
+            catch (IOException e) {
+                handler.fail();
+                throw e;
+            }
         }
     }
 
     private void sendOutput(PrintStream out, OutputFormat format, List<String> fieldNames)
             throws IOException
     {
-        try (OutputHandler handler = createOutputHandler(format, createWriter(out), fieldNames)) {
-            handler.processRows(client);
+        try (Writer writer = createWriter(out);
+                OutputHandler handler = createOutputHandler(format, writer, fieldNames)) {
+            try {
+                handler.processRows(client);
+            }
+            catch (IOException e) {
+                handler.fail();
+                throw e;
+            }
```

Comment:
This try-catch block repeats three times, so maybe it would worth to wrap in a method, to limit code repetition. What do you think?